File: frontend/src/app/models/notification.ts
import { JobApplicationMessage } from "./job_application_message";

export interface Notification {
  id: number;
  userId: number;
  jobApplicationId: string;
  text: string;
  isRead: boolean;
  message: JobApplicationMessage;
  timestamp: Date;
  formattedTimestamp: string;
}

File: JobApplicationMicroservice/src/main/java/com/licenta/jobapplicationmicroservice/business/model/Notification.java
package com.licenta.jobapplicationmicroservice.business.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Notification {

    private Long userId;
    private Long jobApplicationId;
    private String text;
    private LocalDateTime timestamp;
}
